tensorflow/contrib/seq2seq/python/ops/decoder_fn.py

```diff
@@ -0,0 +1,250 @@
+    logits.
+
+    An example of an output function;
+
+    ```
+      tf.variable_scope("decoder") as varscope
+        output_fn = lambda x: layers.linear(x, num_decoder_symbols,
+                                            scope=varscope)
+
+        outputs_train, state_train = seq2seq.dynamic_rnn_decoder(...)
+        logits_train = output_fn(outputs_train)
+
+        varscope.reuse_variables()
+        logits_inference, state_inference = seq2seq.dynamic_rnn_decoder(
+            output_fn=output_fn, ...)
+    ```
+
+    If `None` is supplied it will act as an identity function, which
+    might be wanted when using the RNNCell `OutputProjectionWrapper`.
+
+    encoder_state: The encoded state to initialize the `dynamic_rnn_decoder`.
+    embeddings: The embeddings matrix used for the decoder sized
+    `[num_decoder_symbols, embedding_size]`.
+    start_of_sequence_id: The start of sequence ID in the decoder embeddings.
+    end_of_sequence_id: The end of sequence ID in the decoder embeddings.
+    maximum_length: The maximum allowed of time steps to decode.
+    num_decoder_symbols: The number of classes to decode at each time step.
+    dtype: (default: `dtypes.int32`) The default data type to use when
+    handling integer objects.
+    name: (default: `None`) NameScope for the decoder function;
+      defaults to "simple_decoder_fn_inference"
+
+  Returns:
+    A decoder function with the required interface of `dynamic_rnn_decoder`
+    intended for inference.
+  """
+  with ops.name_scope(name, "simple_decoder_fn_inference",
+                      [output_fn, encoder_state, embeddings,
+                       start_of_sequence_id, end_of_sequence_id,
+                       maximum_length, num_decoder_symbols, dtype]):
+    start_of_sequence_id = ops.convert_to_tensor(start_of_sequence_id, dtype)
+    end_of_sequence_id = ops.convert_to_tensor(end_of_sequence_id, dtype)
+    maximum_length = ops.convert_to_tensor(maximum_length, dtype)
+    num_decoder_symbols = ops.convert_to_tensor(num_decoder_symbols, dtype)
+    encoder_info = nest.flatten(encoder_state)[0]
+    batch_size = encoder_info.get_shape()[0].value
+    if output_fn is None:
+      output_fn = lambda x: x
+    if batch_size is None:
+      batch_size = array_ops.shape(encoder_info)[0]
+
+  def decoder_fn(time, cell_state, cell_input, cell_output, context_state):
+    """ Decoder function used in the `dynamic_rnn_decoder` with the purpose of
+    inference.
+
+    The main difference between this decoder function and the `decoder_fn` in
+    `simple_decoder_fn_train` is how `next_cell_input` is calculated. In this
+    decoder function we calculate the next input by applying an argmax across
+    the feature dimension of the output from the decoder. This approach is
+    taken from popular neural machine translation litterature such as:
+    Bahhdanau et al., 2014 (https://arxiv.org/abs/1409.0473) and Sutskever
+    et al., 2014 (https://arxiv.org/abs/1409.3215).
+
```

Comment:
I think you should replace the last sentence by saying: This is a greedy-search approach.
(Bahdanau et al., 2014) & (Sutskever et al., 2014) use beam-search instead.